.github/workflows/ci.yaml

```diff
@@ -8,177 +8,303 @@ on:
-        uses: ./ci/images/debian10
+      - name: Install helm
+        uses: azure/setup-helm@v1
+
+      - name: Fetch dependencies from cache
+        id: cache-yarn
+        uses: actions/cache@v2
         with:
-          args: ./ci/steps/lint.sh
+          path: "**/node_modules"
+          key: yarn-build-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install dependencies
+        if: steps.cache-yarn.outputs.cache-hit != 'true'
+        run: yarn --frozen-lockfile
 
-  test-unit:
+      - name: Run yarn fmt
+        run: yarn fmt
+        if: success()
+
+      - name: Run yarn lint
+        run: yarn lint
+        if: success()
+
+      - name: Run code-server unit tests
+        run: yarn test:unit
+        if: success()
+
+  build:
+    name: Build
+    needs: prebuild
     runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v1
-      - name: Run unit tests
-        uses: ./ci/images/debian10
-        with:
-          args: ./ci/steps/test-unit.sh
-  test-e2e:
-    needs: linux-amd64
-    runs-on: ubuntu-latest
-    env:
-      PASSWORD: e45432jklfdsab
-      CODE_SERVER_ADDRESS: http://localhost:8080
     steps:
       - uses: actions/checkout@v2
-      - name: Download release packages
-        uses: actions/download-artifact@v2
         with:
-          name: release-packages
-          path: ./release-packages
-      - name: Untar code-server file
-        run: |
-          cd release-packages && tar -xzf code-server*-linux-amd64.tar.gz
-      - uses: microsoft/playwright-github-action@v1
-      - name: Install dependencies and run end-to-end tests
-        run: |
-          ./release-packages/code-server*-linux-amd64/bin/code-server &
-          yarn --frozen-lockfile
-          yarn test:e2e
-      - name: Upload test artifacts
-        if: always()
-        uses: actions/upload-artifact@v2
+          fetch-depth: 0
+
+      - name: Install node.js v12
+        uses: actions/setup-node@v2
         with:
-          name: test-videos
-          path: ./test/e2e/videos
-      - name: Remove release packages and test artifacts
-        run: rm -rf ./release-packages ./test/e2e/videos
+          node-version: "12"
 
-  release:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - name: Run ./ci/steps/release.sh
-        uses: ./ci/images/debian10
+      - name: Fetch dependencies from cache
+        id: cache-yarn
+        uses: actions/cache@v2
         with:
-          args: ./ci/steps/release.sh
+          path: "**/node_modules"
+          key: yarn-build-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install dependencies
+        if: steps.cache-yarn.outputs.cache-hit != 'true'
+        run: yarn --frozen-lockfile
+
+      - name: Build code-server
+        run: yarn build
+
+      # Parse the hash of the latest commit inside lib/vscode
+      # use this to avoid rebuilding it if nothing changed
+      - name: Get latest lib/vscode rev
+        id: vscode-rev
```

Comment:
What's this for?

Reply:
this - the hash of the last commit that touched a file inside `lib/vscode` - is the "key" that I'm using to cache the vscode build - it takes about 5-10 minutes in CI, so if we've built it once before and no files inside changed, just pull the build from the cache and then package it up for the next step.

to check history, I needed to set `fetch-depth: 0` in `actions/checkout`, but a full clone instead of a shallow clone only costs us about three to five seconds right now.

Reply:
Ahh, I see now. That makes sense! Might be worth adding what you wrote as part of your comment so Future us won't forget!